server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -828,4 +835,107 @@ public ClusterState apply(ClusterState state) {
             return builder.build();
         }
     }
+
+    private static boolean validateColumnPositions(Metadata metadata) {
```

Comment:
Having 5 `validateColumnPositions` method variants seems a bit weird.

Can we have something like a walk method to walk over all properties of the metadata and then just visit the positions? Then asserting that there are no duplicates shouldn't be too complicated.

Reply:
@mfussenegger, for the worst case scenario, could these asserts be dropped before merge/backport? And handle it as a separate issue?

Reply:
yep. In any case we shouldn't do it like done here in the PR.

As discussed, the validation should happen closer to where the mistake happens - which would be the inner build methods of the Index/template MetaData builders.

The ClusterState could then check that template/index metadata are consistent, but we can follow up on that as well.

Reply:
Could we remove this for now?